cf-serverd/cf-serverd-functions.c

```diff
@@ -455,8 +455,8 @@ static void CheckFileChanges(EvalContext *ctx, Policy **policy, GenericAgentConf
             UpdateLastPolicyUpdateTime(ctx);
 
             DetectEnvironment(ctx);
+            GenericAgentDiscoverContext(ctx, config);
             KeepHardClasses(ctx);
```

Comment:
What does KeepHardClasses do?
It's important that augments is loaded after environment detection and hard classes are set.

Reply:
Everything `KeepHardClasses` does is already done by `GenericAgentDiscoverContext`. I will try to remove it.